DevelopGitHub/GitHubExtension/GitHubExtension.Statistics.WebApi/Controllers/StatisticController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Web.Http;
+using GitHubExtension.Statistics.WebApi.CommunicationModels;
+using GitHubExtension.Statistics.WebApi.Constant;
+using GitHubExtension.Statistics.WebApi.Mappers.Identity;
+using GitHubExtension.Statistics.WebApi.Queries.Interfaces;
+using Microsoft.AspNet.Identity;
+
+namespace GitHubExtension.Statistics.WebApi.Controllers
+{
+    public class StatisticController : ApiController
+    {
+        #region fields
+        private readonly IStatisticsQuery _statisticsQuery;
+        #endregion
+
+        public StatisticController(IStatisticsQuery statisticsQuery)
+        {
+            this._statisticsQuery = statisticsQuery;
+        }
+
+        #region methods
+        [HttpGet]
+        [Route(StatisticsRouteConstants.GetUserCommits)]
+        public async Task<IHttpActionResult> GetCommitsForUser()
+        {
+            string token = User.Identity.GetExternalAccessToken();
+            string userName = User.Identity.GetUserName();
+            GraphModel graph = await _statisticsQuery.GraphCreation(userName, token);
+
+            return Ok(graph);
+        }
+
+        [HttpGet]
+        [Route(StatisticsRouteConstants.GetRepoByName)]
+        public async Task<IHttpActionResult> GetRepo([FromUri] string name)
```

Comment:
I like such small piceces of code with loads data. Because it can be tuned and optimize in independent manner.